src/main/java/com/ibm/cloud/sdk/core/security/TokenRequestBasedAuthenticator.java

```diff
@@ -211,9 +239,23 @@ public void authenticate(Builder builder) {
   public String getToken() {
     String token;
 
-    if (this.tokenData == null || !this.tokenData.isTokenValid()) {
-      // request new token
-      this.tokenData = requestToken();
+    if (this.tokenData == null) {
+      this.tokenData = synchronizedRequestToken();
+    } else if (!this.tokenData.isTokenValid()) {
+
+      // Token is "invalid" but it's not expired, meaning it's in our buffer zone. We can still return the stored
+      // token while we go ahead and get a new token in the background.
+      if (!this.tokenData.isTokenExpired()) {
```

Comment:
I don't think you need this check here.  If the token needs to be refreshed, we always want to refresh.

Reply:
Yeah looking at it again I think you're right. I initially had the check there to make sure that we don't return the stored access token if it's actually completely expired, but I don't think the refresh logic needs to be included in that.